packages/react-native-aztec/src/AztecView.js

```diff
@@ -237,14 +237,12 @@ class AztecView extends Component {
 	}
 
 	_onAztecFocus( event ) {
-		// IMPORTANT: the onFocus events from Aztec are thrown away on Android as these are handled by onPress() in the upper level.
+		// IMPORTANT: the onFocus events from Aztec are thrown away as these are handled by onPress() in the upper level.
 		// It's necessary to do this otherwise onFocus may be set by `{...otherProps}` and thus the onPress + onFocus
 		// combination generate an infinite loop as described in https://github.com/wordpress-mobile/gutenberg-mobile/issues/302
-		// For iOS, this is necessary to let the system know when Aztec was focused programatically.
+		// For iOS, this is necessary to let the system know the current caret data.
 		if ( Platform.OS === 'ios' ) {
 			this.updateCaretData( event );
-
-			this._onPress( event );
```

Comment:
While I do believe invoking `_onPress` is unnecessary and problematic, I believe there may still be a need to communicate the focus change to the native Aztec module. This is referenced in #44988. 

https://github.com/WordPress/gutenberg/blob/fda398ea647389c5abd28093f3d5035c1a1d6c56/packages/react-native-aztec/src/AztecView.js#L245-L251

Without communicating the change of focus, following the steps outlined in https://github.com/wordpress-mobile/gutenberg-mobile/issues/702 will result in the keyboard hide button no longer functioning. 

**Are you able to replicate this non-functioning keyboard dismiss button issue as well?** I am only able to reproduce in a WPiOS prototype build; when using the Demo editor, the keyboard merely never displays when dismissing the link bottom sheet. 

<details><summary>Non-functioning keyboard dismissal button</summary>

https://github.com/WordPress/gutenberg/assets/438664/9574884e-789c-4ca8-879d-eee86dbe49bb

</details>

The approach explored in #44988 was to invoke a newly define `AztecInputState.focusInput` rather than `_onPress` to communicate the change of focus without starting a focus loop via `_onPress`. 

https://github.com/WordPress/gutenberg/blob/fda398ea647389c5abd28093f3d5035c1a1d6c56/packages/react-native-aztec/src/AztecView.js#L261-L268

Additionally or alternatively — not sure which is true at the moment — we may need to reintroduce [explicitly dismissing the keyboard](https://github.com/WordPress/gutenberg/pull/44988/files#r1012217403). IIRC the explicit keyboard dismissal addition was strictly related to simultaneously selecting two blocks; focusing a rich text field in one block while selecting a different block. 

I am unable to reproduce that scenario in this branch, so the explicit keyboard dismissal may not be necessary. However, it could be that I am unable to reproduce the scenario due to the lack of communicating the new focus state to Aztec outlined above.

Reply:
> Are you able to replicate this non-functioning keyboard dismiss button issue as well? 
Yes, but only for the steps outlined in that PR:

1. Add a link
2. Dismiss the bottom sheet
3. Attempt to hide the keyboard
4. Keyboard does not hide

In all other inputs I tested, the keyboard hide button was working as expected.

Reply:
Hey @dcalhoun 👋  thank you so much for the review! Nice catch about link picker, after several tests, I went ahead by cherry-picking your approach into this branch and doing some tweaks, I hope you don't mind bringing them into this branch, we could also merge my changes into your PR if it's updated with `trunk`. Let me know either way what you think!

>Are you able to replicate this non-functioning keyboard dismiss button issue as well? I am only able to reproduce in a WPiOS prototype build; when using the Demo editor, the keyboard merely never displays when dismissing the link bottom sheet.

Yes I was able to reproduce it constantly, as for the demo app, it was due to another issue that was already addressed in https://github.com/WordPress/gutenberg/pull/53270

> The approach explored in https://github.com/WordPress/gutenberg/pull/44988 was to invoke a newly define AztecInputState.focusInput rather than _onPress to communicate the change of focus without starting a focus loop via _onPress.

Yeah, I think this is the best way to go to avoid triggering other things that could cause a focus loop, we just need the internal ref to be updated.

> Additionally or alternatively — not sure which is true at the moment — we may need to reintroduce [explicitly dismissing the keyboard](https://github.com/WordPress/gutenberg/pull/44988/files#r1012217403). IIRC the explicit keyboard dismissal addition was strictly related to simultaneously selecting two blocks; focusing a rich text field in one block while selecting a different block.

I wasn't able to reproduce this so I didn't include it for now.

Reply:
> Nice catch about the link picker, after several tests, I went ahead by cherry-picking your approach into this branch and doing some tweaks, I hope you don't mind bringing them into this branch, we could also merge my changes into your PR if it's updated with `trunk`. Let me know either way what you think!

Cherry picking is a perfectly fine approach from my perspective. 

> I wasn't able to reproduce [focusing a rich text field outside of the selected block] so I didn't include it for now.

Below is a video of me reproducing the issue with the `pr21209-94fabdf` build in https://github.com/wordpress-mobile/WordPress-iOS/pull/21209#issuecomment-1658561718. I reproduce it on my first try when recording the UI. 🥇😄 It took a few more tries when recording the interaction. 

| UI recording | Interaction recording |
| - | - |
| <video src="https://github.com/WordPress/gutenberg/assets/438664/5c0e92ce-76f6-4ede-b093-cfac1530da08" /> | <video src="https://github.com/WordPress/gutenberg/assets/438664/ed01fd7e-fc79-40af-bc11-88bdec15755d" /> |

Reply:
>Below is a video of me reproducing the issue with the pr21209-94fabdf build in https://github.com/wordpress-mobile/WordPress-iOS/pull/21209#issuecomment-1658561718. I reproduce it on my first try when recording the UI. 🥇😄 It took a few more tries when recording the interaction.

Those videos definitely helped! Thanks!

[I created a new build](https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/releases/5610) with the latest commit in this branch, it should address that issue and still keep the other bugs fixed (hopefully 😄)